### PR TITLE
Adds stories/documentation for useCheckoutPricing

### PR DESCRIPTION
**Pull request by dbrudner**

### CODE REVIEW

File: docs/3-reference/useCheckoutPricingReturn.stories.mdx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import { Subtitle } from '@storybook/addon-docs/blocks';
+import { PropsTable } from '@storybook/components';
+
+<Meta title="Reference/UseCheckoutPricingReturn" />
+
+# UseCheckoutPricingReturn
+
+<Subtitle slot={() => 'useCheckoutPricing() return value'} />
+
```
Comment (chrissrogers) [inline docs/3-reference/useCheckoutPricingReturn.stories.mdx]: I'm thinking a brief description here will help provide context for those just clicking around, with a link back to the `useCheckoutPricing` section for a full overview.

File: docs/3-reference/useCheckoutPricingInput.stories.mdx [hunk 2]
```diff
@@ -0,0 +1,217 @@
+import { Subtitle } from '@storybook/addon-docs/blocks';
+import { PropsTable } from '@storybook/components';
+
+<Meta title="Reference/UseCheckoutPricingInput" />
+
+# UseCheckoutPricingInput
+
+<Subtitle slot={() => 'To interact with useCheckoutPricing'} />
+
+A plain object used to set initial values for [useCheckoutPricing][use-checkout-pricing-input] and to update those values if needed with [setCheckoutPricing][use-checkout-pricing-set-pricing].
```
Comment (chrissrogers) [inline docs/3-reference/useCheckoutPricingInput.stories.mdx]: Thoughts on adding a note that overall usage is covered in the `useCheckoutPricing` section?

File: docs/3-reference/useCheckoutPricingReturn.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    },
+    {
+      name: 'loading',
+      type: { summary: 'boolean' },
+      description: 'Describes the current loading state of useCheckoutPricing',
+    },
+  ]}
+/>
+
+### SetCheckoutPricing
```
Comment (chrissrogers) [inline docs/3-reference/useCheckoutPricingReturn.stories.mdx]: ed: capitalization is incorrect.

File: docs/2-hooks/useCheckoutPricing.stories.mdx [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+  }
+  ...
+}
+```
+
+Errors will always take the shape of a [Recurly.js error][recurly-error].
+
+## Demo
+
+For a full example of `useCheckoutPricing`, see our [interactive demo][interactive-demo].
```
Comment (chrissrogers) [inline docs/2-hooks/useCheckoutPricing.stories.mdx]: Just noting that this is implemented but not yet published to CodeSandbox.

File: docs/2-hooks/useCheckoutPricing.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+[react-hooks]: https://reactjs.org/docs/hooks-intro.html
+[rjs-pricing]: https://developers.recurly.com/reference/recurly-js/#pricing
+[react-use-state]: https://reactjs.org/docs/hooks-state.html
+[recurly-error]: https://developers.recurly.com/reference/recurly-js/#errors
+[interactive-demo]: /?path=/docs/introduction-interactive-demo--page
+[set-checkout-pricing]: /?path=/docs/reference-usecheckoutpricingreturn--page#setcheckoutpricing)
+[use-checkout-pricing-state]: /?path=/docs/reference-usecheckoutpricingreturn--page#usecheckoutpricingstate
+[use-checkout-pricing-return]: /?path=/docs/reference-usecheckoutpricingreturn--page
+[use-checkout-pricing-input]: /?path=/docs/reference-usecheckoutpricinginput--page
+[use-checkout-pricing-updating]: /?path=/docs/reference-usecheckoutpricinginput--page#updating
```
Comment (chrissrogers) [inline docs/2-hooks/useCheckoutPricing.stories.mdx]: This is an amazing implementation guide 👏

File: docs/2-hooks/useCheckoutPricing.stories.mdx [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+<Subtitle slot={() => 'To create a pricing preview'} />
+
+`import { useCheckoutPricing } from '@recurly/react-recurly';`
+
+A [custom hook][react-hooks] for interacting with [Recurly.js' pricing API][rjs-pricing] for providing users with an estimate of a purchase before they check out.
+
+To initialize, call `useCheckoutPricing` [with an initial checkout pricing input][use-checkout-pricing-input].
+
+```js
+export default function PricingPreview() {
```
Comment (chrissrogers) [inline docs/2-hooks/useCheckoutPricing.stories.mdx]: lint nit: `PricingPreview ()`

File: docs/2-hooks/useCheckoutPricing.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+  const initialPricingInput = {
+    subscriptions: [
+      {
+        plan: 'my-plan',
+      },
+    ],
```
Comment (chrissrogers) [inline docs/2-hooks/useCheckoutPricing.stories.mdx]: lint nit: unnecessary commas

File: docs/2-hooks/useCheckoutPricing.stories.mdx [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+  };
+
+  const [{ price, loading }, setCheckoutPricing] = useCheckoutPricing(initialPricingInput);
+
+  if (!loading) {
+    return <div>{price.now && price.now.subtotal}</div>
```
Comment (chrissrogers) [inline docs/2-hooks/useCheckoutPricing.stories.mdx]:
```suggestion
    return <div>{price.now?.subtotal}</div>
```

File: docs/2-hooks/useCheckoutPricing.stories.mdx [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+
+## Updating
+
+The [setCheckoutPricing][set-checkout-pricing] function can be used to update the checkout pricing input if needed with a new [useCheckoutPricingInput][use-checkout-pricing-input].
+
+For example, if a customer wishes to choose a different plan for their subscription on a checkout page and the price preview needs to be updated, calling `setCheckoutPricing` with new inputs will update its associated `useCheckoutPricingState` and trigger a re-render.
+
+```js
+export default function PricingPreview() {
+  const [plan, setPlan] = useState("")
```
Comment (chrissrogers) [inline docs/2-hooks/useCheckoutPricing.stories.mdx]: lint nit: missing terminal semicolon

**Comment by chrissrogers** [comment]
I think adding a section to the README is appropriate.

After the `## Implementation Guide` an `## Additional Usage` section with a short description of `useCheckoutPricing` with a link to the hook doc will be super useful.

File: docs/2-hooks/useCheckoutPricing.stories.mdx (hunk 8 above)
Comment (dbrudner) [reply docs/2-hooks/useCheckoutPricing.stories.mdx]: Ah yeah this is a mistake. It should just be `price.now.subtotal`, the loading boolean protects the type error from happening.

File: docs/3-reference/useCheckoutPricingInput.stories.mdx [hunk 10]
```diff
@@ -0,0 +1,217 @@
+import { Subtitle } from '@storybook/addon-docs/blocks';
+import { PropsTable } from '@storybook/components';
+
+<Meta title="Reference/UseCheckoutPricingInput" />
```
[the remark this answers is not in the document]
Comment (dbrudner) [reply docs/3-reference/useCheckoutPricingInput.stories.mdx]: +1

File: docs/3-reference/useCheckoutPricingReturn.stories.mdx (hunk 1 above)
Comment (dbrudner) [reply docs/3-reference/useCheckoutPricingReturn.stories.mdx]: Agreed, good call

File: docs/2-hooks/useCheckoutPricing.stories.mdx (hunk 9 above)
Comment (dbrudner) [reply docs/2-hooks/useCheckoutPricing.stories.mdx]: +1, I have to clean up these code examples a bit

File: docs/3-reference/useCheckoutPricingReturn.stories.mdx (hunk 3 above)
Comment (dbrudner) [reply docs/3-reference/useCheckoutPricingReturn.stories.mdx]: I capitalized all the headers for the types. Would camelcasing everything be more appropriate?

**Pull request merged**